sunpy/coordinates/metaframes.py

```diff
@@ -186,6 +187,16 @@ def __init__(self, *args, **kwargs):
             kwargs['duration'] = (rotated_time - kwargs['base'].obstime).to('day')
             kwargs.pop('rotated_time')
 
+        if 'duration' in kwargs:
+            duration = kwargs['duration']
+            if isinstance(duration, TimeDelta):
+                kwargs['duration'] = duration.to(u.day)
+            elif isinstance(duration, u.Quantity):
+                if duration.unit != u.day:
+                    kwargs['duration'] = duration.to(u.day)
+            else:
+                raise ValueError("`duration` must be a `TimeDelta` or `Quantity` object.")
```

Comment:
There should be a unit test to check this line.

Reply:
Yes, I will work on that